src/bun.js/node/types.zig

```diff
@@ -2086,8 +2086,9 @@ pub const Process = struct {
             return;
         }
 
+        vm.exit_handler.exit_code = code;
         vm.onExit();
-        bun.Global.exit(code);
+        bun.Global.exit(vm.exit_handler.exit_code);
```

Comment:
What was the purpose of adding the extra pointer lookup here? If it's to account for the user changing the exit code inside of onExit (assuming Node supports that), then maybe this should be moved into a method on vm?